Rename WKSafeBrowsingWarning to WKWarningView

sysrqb opened the pull request:
#### fc46bbbaea29d6bc0c9d342e45c5e24fb3da0a63
<pre>
Rename WKSafeBrowsingWarning to WKWarningView
<a href="https://bugs.webkit.org/show_bug.cgi?id=277442">https://bugs.webkit.org/show_bug.cgi?id=277442</a>
<a href="https://rdar.apple.com/132924925">rdar://132924925</a>

Reviewed by Alex Christensen.

Rename WKSafeBrowsingWarning, and associated state, to a more generic name.
This will be useful in future changes.

No other behavioral changes.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _showWarningView:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarning:completionHandler:]):
(-[WKWebView _clearWarningView]):
(-[WKWebView _clearSafeBrowsingWarning]):
(-[WKWebView _clearWarningViewIfForMainFrameNavigation]):
(-[WKWebView _clearSafeBrowsingWarningIfForMainFrameNavigation]):
(-[WKWebView _showWarningViewWithTitle:warning:details:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarningWithTitle:warning:details:completionHandler:]):
(-[WKWebView _showWarningViewWithURL:title:warning:details:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarningWithURL:title:warning:details:completionHandler:]):
(-[WKWebView _showWarningViewWithURL:title:warning:detailsWithLinks:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarningWithURL:title:warning:detailsWithLinks:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView layoutSubviews]):
(-[WKWebView _computedObscuredInsetForWarningView]):
(-[WKWebView safeAreaInsetsDidChange]):
(-[WKWebView _setObscuredInsets:]):
(-[WKWebView _safeBrowsingWarning]):
(-[WKWebView _computedObscuredInsetForSafeBrowsingWarning]): Deleted.
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView setFrameSize:]):
(-[WKWebView _safeBrowsingWarning]):
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.h: Renamed from Source/WebKit/UIProcess/Cocoa/WKSafeBrowsingWarning.h.
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm: Renamed from Source/WebKit/UIProcess/Cocoa/WKSafeBrowsingWarning.mm.
(colorForItem):
(-[WKSafeBrowsingExclamationPoint drawRect:]):
(makeButton):
(-[_WKWarningViewBox setWarningViewBackgroundColor:]):
(-[_WKWarningView initWithFrame:safeBrowsingWarning:completionHandler:]):
(-[_WKWarningView addContent]):
(-[_WKWarningView showDetailsClicked]):
(-[_WKWarningView clickedOnLink:]):
(-[_WKWarningViewTextView initWithAttributedString:forWarning:]):
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::showSafeBrowsingWarning):
(WebKit::PageClientImpl::hasSafeBrowsingWarning const):
(WebKit::PageClientImpl::clearSafeBrowsingWarning):
(WebKit::PageClientImpl::clearSafeBrowsingWarningIfForMainFrameNavigation):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::showWarningView):
(WebKit::WebViewImpl::clearWarningView):
(WebKit::WebViewImpl::clearWarningViewIfForMainFrameNavigation):
(WebKit::WebViewImpl::setFrameSize):
(WebKit::WebViewImpl::accessibilityAttributeValue):
(WebKit::WebViewImpl::nativeMouseEventHandlerInternal):
(WebKit::WebViewImpl::showSafeBrowsingWarning): Deleted.
(WebKit::WebViewImpl::clearSafeBrowsingWarning): Deleted.
(WebKit::WebViewImpl::clearSafeBrowsingWarningIfForMainFrameNavigation): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/281715@main">https://commits.webkit.org/281715@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/618070e1907e2f17e6088c02f999a65727c1daf4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60649 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40008 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13225 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64580 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11196 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47684 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11471 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49051 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7775 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62682 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37262 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52538 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29905 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33952 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10109 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55808 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10067 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66309 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4593 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9914 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56421 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4614 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52511 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56598 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3807 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9138 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35813 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36895 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37988 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36640 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
```diff
@@ -429,9 +429,15 @@ struct PerWebProcessState {
+- (void)_clearWarningViewIfForMainFrameNavigation;
 - (void)_clearSafeBrowsingWarningIfForMainFrameNavigation;
+#if PLATFORM(IOS_FAMILY)
+- (UIView *)_safeBrowsingWarning;
```
This is suspicious.  Shouldn't we have an NSView property on macOS?

achristensen07 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
```diff
@@ -118,7 +118,7 @@ class ViewGestureController;
 @class WKContentView;
 @class WKPasswordView;
-@class WKSafeBrowsingWarning;
+@class WKWarningView;
```
Could we name this _WKWarningView because it's not intended to be used by third parties?

sysrqb commented on Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm:
```diff
@@ -185,7 +185,7 @@ - (void)drawRect:(RectType)rect
     auto square = CGRectMake(0, 0, exclamationPointSize, exclamationPointSize);
     [[UIBezierPath bezierPathWithRect:square] fill];
-    
+
```
Style checker wasn't happy with this pre-existing whitespace. Same with the other similar changes.

Commit:
Rename WKSafeBrowsingWarning to WKWarningView
https://bugs.webkit.org/show_bug.cgi?id=277442
rdar://132924925

Reviewed by Alex Christensen.

Rename WKSafeBrowsingWarning, and associated state, to a more generic name.
This will be useful in future changes.

No other behavioral changes.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _showWarningView:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarning:completionHandler:]):
(-[WKWebView _clearWarningView]):
(-[WKWebView _clearSafeBrowsingWarning]):
(-[WKWebView _clearWarningViewIfForMainFrameNavigation]):
(-[WKWebView _clearSafeBrowsingWarningIfForMainFrameNavigation]):
(-[WKWebView _showWarningViewWithTitle:warning:details:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarningWithTitle:warning:details:completionHandler:]):
(-[WKWebView _showWarningViewWithURL:title:warning:details:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarningWithURL:title:warning:details:completionHandler:]):
(-[WKWebView _showWarningViewWithURL:title:warning:detailsWithLinks:completionHandler:]):
(-[WKWebView _showSafeBrowsingWarningWithURL:title:warning:detailsWithLinks:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView layoutSubviews]):
(-[WKWebView _computedObscuredInsetForWarningView]):
(-[WKWebView safeAreaInsetsDidChange]):
(-[WKWebView _setObscuredInsets:]):
(-[WKWebView _safeBrowsingWarning]):
(-[WKWebView _computedObscuredInsetForSafeBrowsingWarning]): Deleted.
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView setFrameSize:]):
(-[WKWebView _safeBrowsingWarning]):
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.h: Renamed from Source/WebKit/UIProcess/Cocoa/WKSafeBrowsingWarning.h.
* Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm: Renamed from Source/WebKit/UIProcess/Cocoa/WKSafeBrowsingWarning.mm.
(colorForItem):
(-[WKSafeBrowsingExclamationPoint drawRect:]):
(makeButton):
(-[_WKWarningViewBox setWarningViewBackgroundColor:]):
(-[_WKWarningView initWithFrame:safeBrowsingWarning:completionHandler:]):
(-[_WKWarningView addContent]):
(-[_WKWarningView showDetailsClicked]):
(-[_WKWarningView clickedOnLink:]):
(-[_WKWarningViewTextView initWithAttributedString:forWarning:]):
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::showSafeBrowsingWarning):
(WebKit::PageClientImpl::hasSafeBrowsingWarning const):
(WebKit::PageClientImpl::clearSafeBrowsingWarning):
(WebKit::PageClientImpl::clearSafeBrowsingWarningIfForMainFrameNavigation):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::showWarningView):
(WebKit::WebViewImpl::clearWarningView):
(WebKit::WebViewImpl::clearWarningViewIfForMainFrameNavigation):
(WebKit::WebViewImpl::setFrameSize):
(WebKit::WebViewImpl::accessibilityAttributeValue):
(WebKit::WebViewImpl::nativeMouseEventHandlerInternal):
(WebKit::WebViewImpl::showSafeBrowsingWarning): Deleted.
(WebKit::WebViewImpl::clearSafeBrowsingWarning): Deleted.
(WebKit::WebViewImpl::clearSafeBrowsingWarningIfForMainFrameNavigation): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/281715@main
